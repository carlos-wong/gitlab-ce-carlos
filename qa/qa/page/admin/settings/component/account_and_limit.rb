# frozen_string_literal: true

module QA
  module Page
    module Admin
      module Settings
        module Component
          class AccountAndLimit < Page::Base
            view 'app/views/admin/application_settings/_account_and_limit.html.haml' do
              element :receive_max_input_size_field
              element :save_changes_button
            end

            def set_max_file_size(size)
              fill_element :receive_max_input_size_field, size
            end

            def save_settings
              click_element :save_changes_button
            end
          end
        end
      end
    end
  end
end
