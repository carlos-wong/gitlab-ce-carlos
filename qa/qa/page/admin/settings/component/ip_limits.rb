# frozen_string_literal: true

module QA
  module Page
    module Admin
      module Settings
        module Component
          class IpLimits < Page::Base
            view 'app/views/admin/application_settings/_ip_limits.html.haml' do
              element :throttle_unauthenticated_checkbox
              element :throttle_authenticated_api_checkbox
              element :throttle_authenticated_web_checkbox
              element :save_changes_button
            end

            def enable_throttles
              check_element :throttle_unauthenticated_checkbox
              check_element :throttle_authenticated_api_checkbox
              check_element :throttle_authenticated_web_checkbox
            end

            def save_settings
              click_element :save_changes_button
            end
          end
        end
      end
    end
  end
end
