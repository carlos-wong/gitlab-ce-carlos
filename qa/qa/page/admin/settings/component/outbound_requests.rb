# frozen_string_literal: true

module QA
  module Page
    module Admin
      module Settings
        module Component
          class OutboundRequests < Page::Base
            view 'app/views/admin/application_settings/_outbound.html.haml' do
              element :allow_requests_from_services_checkbox
              element :save_changes_button
            end

            def allow_requests_to_local_network_from_services
              check_allow_requests_to_local_network_from_services_checkbox
              click_save_changes_button
            end

            private

            def check_allow_requests_to_local_network_from_services_checkbox
              check_element :allow_requests_from_services_checkbox
            end

            def click_save_changes_button
              click_element :save_changes_button
            end
          end
        end
      end
    end
  end
end
