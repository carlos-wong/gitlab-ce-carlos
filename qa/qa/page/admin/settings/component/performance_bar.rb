# frozen_string_literal: true

module QA
  module Page
    module Admin
      module Settings
        module Component
          class PerformanceBar < Page::Base
            view 'app/views/admin/application_settings/_performance_bar.html.haml' do
              element :enable_performance_bar_checkbox
              element :save_changes_button
            end

            def enable_performance_bar
              click_element :enable_performance_bar_checkbox
              Capybara.current_session.driver.browser.manage.add_cookie(name: 'perf_bar_enabled', value: 'true')
            end

            def save_settings
              click_element :save_changes_button
            end
          end
        end
      end
    end
  end
end
