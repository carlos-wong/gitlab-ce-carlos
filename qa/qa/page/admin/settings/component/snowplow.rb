# frozen_string_literal: true

module QA
  module Page
    module Admin
      module Settings
        module Component
          class Snowplow < Page::Base
            include QA::Page::Settings::Common

            view 'app/views/admin/application_settings/_snowplow.html.haml' do
              element :snowplow_settings_content
              element :snowplow_enabled_checkbox
              element :save_changes_button
            end

            def enable_snowplow_tracking
              expand_content(:snowplow_settings_content) do
                check_snowplow_enabled_checkbox
                click_save_changes_button
              end
            end

            def disable_snowplow_tracking
              expand_content(:snowplow_settings_content) do
                uncheck_snowplow_enabled_checkbox
                click_save_changes_button
              end
            end

            private

            def check_snowplow_enabled_checkbox
              check_element(:snowplow_enabled_checkbox, true)
            end

            def uncheck_snowplow_enabled_checkbox
              uncheck_element(:snowplow_enabled_checkbox, true)
            end

            def click_save_changes_button
              click_element :save_changes_button
            end
          end
        end
      end
    end
  end
end
