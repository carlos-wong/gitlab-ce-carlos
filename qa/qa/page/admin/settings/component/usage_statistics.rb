# frozen_string_literal: true

module QA
  module Page
    module Admin
      module Settings
        module Component
          class UsageStatistics < Page::Base
            view 'app/views/admin/application_settings/_usage.html.haml' do
              element :enable_usage_data_checkbox
            end

            def has_disabled_usage_data_checkbox?
              has_element?(:enable_usage_data_checkbox, disabled: true, visible: false)
            end
          end
        end
      end
    end
  end
end
