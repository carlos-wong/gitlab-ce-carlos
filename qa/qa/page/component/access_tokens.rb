# frozen_string_literal: true

module QA
  module Page
    module Component
      module AccessTokens
        extend QA::Page::PageConcern

        def self.included(base)
          super

          base.view 'app/assets/javascripts/access_tokens/components/expires_at_field.vue' do
            element :expiry_date_field
          end

          base.view 'app/views/shared/access_tokens/_form.html.haml' do
            element :access_token_name_field
            element :create_token_button
          end

          base.view 'app/views/shared/tokens/_scopes_form.html.haml' do
            element :api_label, '#{scope}_label' # rubocop:disable QA/ElementWithPattern, Lint/InterpolationCheck
          end

          base.view 'app/views/shared/access_tokens/_created_container.html.haml' do
            element :created_access_token
          end

          base.view 'app/views/shared/access_tokens/_table.html.haml' do
            element :revoke_button
          end
        end

        def fill_token_name(name)
          fill_element(:access_token_name_field, name)
        end

        def check_api
          click_element(:api_label)
        end

        def click_create_token_button
          click_element(:create_token_button)
        end

        def created_access_token
          find_element(:created_access_token, wait: 30).value
        end

        def fill_expiry_date(date)
          date = date.to_s if date.is_a?(Date)
          Date.strptime(date, '%Y-%m-%d') rescue ArgumentError raise "Expiry date must be in YYYY-MM-DD format"

          fill_element(:expiry_date_field, date)
        end

        def has_token_row_for_name?(token_name)
          page.has_css?('tr', text: token_name, wait: 1.0)
        end

        def first_token_row_for_name(token_name)
          page.find('tr', text: token_name, match: :first, wait: 1.0)
        end

        def revoke_first_token_with_name(token_name)
          within first_token_row_for_name(token_name) do
            accept_confirm do
              click_element(:revoke_button)
            end
          end
        end
      end
    end
  end
end
