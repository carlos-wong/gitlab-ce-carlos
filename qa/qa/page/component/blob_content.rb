# frozen_string_literal: true

module QA
  module Page
    module Component
      module BlobContent
        extend QA::Page::PageConcern

        def self.included(base)
          super

          base.view 'app/assets/javascripts/blob/components/blob_header_filepath.vue' do
            element :file_title_content
          end

          base.view 'app/assets/javascripts/blob/components/blob_content.vue' do
            element :blob_viewer_file_content
          end

          base.view 'app/assets/javascripts/blob/components/blob_header_default_actions.vue' do
            element :default_actions_container
            element :copy_contents_button
          end

          base.view 'app/assets/javascripts/vue_shared/components/source_viewer/source_viewer.vue' do
            element :blob_viewer_file_content
          end
        end

        def has_file?(name)
          has_file_name?(name)
        end

        def has_no_file?(name)
          has_no_file_name?(name)
        end

        def has_file_name?(file_name, file_number = nil)
          within_file_by_number(:file_title_content, file_number) { has_text?(file_name) }
        end

        def has_no_file_name?(file_name)
          within_element(:file_title_content) do
            has_no_text?(file_name)
          end
        end

        def has_file_content?(file_content, file_number = nil)
          within_file_by_number(:blob_viewer_file_content, file_number) { has_text?(file_content) }
        end

        def has_no_file_content?(file_content)
          within_element(:blob_viewer_file_content) do
            has_no_text?(file_content)
          end
        end

        def has_normalized_ws_text?(text, wait: Capybara.default_max_wait_time)
          if has_element?(:blob_viewer_file_content, wait: 1)
            # The blob viewer renders line numbers and whitespace in a way that doesn't match the source file
            # This isn't a visual validation test, so we ignore line numbers and whitespace
            find_element(:blob_viewer_file_content, wait: 0).text.gsub(/^\d+\s|\s*/, '')
              .start_with?(text.gsub(/\s*/, ''))
          else
            has_text?(text.gsub(/\s+/, " "), wait: wait)
          end
        end

        def click_copy_file_contents(file_number = nil)
          within_file_by_number(:default_actions_container, file_number) { click_element(:copy_contents_button) }
        end

        private

        def within_file_by_number(element, file_number)
          if file_number
            within_element_by_index(element, file_number - 1) { yield }
          else
            within_element(element) { yield }
          end
        end
      end
    end
  end
end
