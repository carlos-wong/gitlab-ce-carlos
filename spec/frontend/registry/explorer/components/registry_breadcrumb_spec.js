import { shallowMount } from '@vue/test-utils';

import component from '~/registry/explorer/components/registry_breadcrumb.vue';

describe('Registry Breadcrumb', () => {
  let wrapper;
  const nameGenerator = jest.fn();

  const crumb = {
    classList: ['foo', 'bar'],
    tagName: 'div',
    innerHTML: 'baz',
    querySelector: jest.fn(),
    children: [
      {
        tagName: 'a',
        classList: ['foo'],
      },
    ],
  };

  const querySelectorReturnValue = {
    classList: ['js-divider'],
    tagName: 'svg',
    innerHTML: 'foo',
  };

  const crumbs = [crumb, { ...crumb, innerHTML: 'foo' }, { ...crumb, classList: ['baz'] }];

  const routes = [
    { name: 'foo', meta: { nameGenerator, root: true } },
    { name: 'baz', meta: { nameGenerator } },
  ];

  const findDivider = () => wrapper.find('.js-divider');
  const findRootRoute = () => wrapper.find({ ref: 'rootRouteLink' });
  const findChildRoute = () => wrapper.find({ ref: 'childRouteLink' });
  const findLastCrumb = () => wrapper.find({ ref: 'lastCrumb' });

  const mountComponent = $route => {
    wrapper = shallowMount(component, {
      propsData: {
        crumbs,
      },
      stubs: {
        'router-link': { name: 'router-link', template: '<a><slot></slot></a>', props: ['to'] },
      },
      mocks: {
        $route,
        $router: {
          options: {
            routes,
          },
        },
      },
    });
  };

  beforeEach(() => {
    nameGenerator.mockClear();
    crumb.querySelector = jest.fn();
  });

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('when is rootRoute', () => {
    beforeEach(() => {
      mountComponent(routes[0]);
    });

    it('renders', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('contains a router-link for the child route', () => {
      expect(findChildRoute().exists()).toBe(true);
    });

    it('the link text is calculated by nameGenerator', () => {
      expect(nameGenerator).toHaveBeenCalledWith(routes[0]);
      expect(nameGenerator).toHaveBeenCalledTimes(1);
    });
  });

  describe('when is not rootRoute', () => {
    beforeEach(() => {
      crumb.querySelector.mockReturnValue(querySelectorReturnValue);
      mountComponent(routes[1]);
    });

    it('renders a divider', () => {
      expect(findDivider().exists()).toBe(true);
    });

    it('contains a router-link for the root route', () => {
      expect(findRootRoute().exists()).toBe(true);
    });

    it('contains a router-link for the child route', () => {
      expect(findChildRoute().exists()).toBe(true);
    });

    it('the link text is calculated by nameGenerator', () => {
      expect(nameGenerator).toHaveBeenCalledWith(routes[1]);
      expect(nameGenerator).toHaveBeenCalledTimes(2);
    });
  });

  describe('last crumb', () => {
    const lastChildren = crumb.children[0];
    beforeEach(() => {
      nameGenerator.mockReturnValue('foo');
      mountComponent(routes[0]);
    });

    it('has the same tag as the last children of the crumbs', () => {
      expect(findLastCrumb().is(lastChildren.tagName)).toBe(true);
    });

    it('has the same classes as the last children of the crumbs', () => {
      expect(findLastCrumb().classes()).toEqual(lastChildren.classList);
    });

    it('has a link to the current route', () => {
      expect(findChildRoute().props('to')).toEqual({ to: routes[0].name });
    });

    it('the link has the correct text', () => {
      expect(findChildRoute().text()).toEqual('foo');
    });
  });
});
