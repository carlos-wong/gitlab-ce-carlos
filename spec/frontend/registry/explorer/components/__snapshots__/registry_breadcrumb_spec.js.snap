// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registry Breadcrumb when is rootRoute renders 1`] = `
<ul>
  <li
    class="foo bar"
  >
    baz
  </li>
  <li
    class="foo bar"
  >
    foo
  </li>
   
  <!---->
   
  <li>
    <a
      class="foo"
    >
      <a>
        
      </a>
    </a>
  </li>
</ul>
`;
